core/py/src/braintrust_core/legacy_api.py

```diff
@@ -0,0 +1,29 @@
+import json
+
+
+def patch_legacy_dataset(r):
+    row = {**r}
+    output = row.pop("output", None)
+    if "expected" not in row:
```

Comment:
nit: it might read more clearly to write it as
```
if "output" in row and "expected" not in row:
    row["expected"] = row.pop("output")
```

just so the reader doesn't have to understand the implicit assertion that the row cannot contain both output and expected